airflow/example_dags/example_branch_operator_decorator.py

```diff
@@ -16,10 +16,12 @@
 # specific language governing permissions and limitations
 # under the License.
 
-"""Example DAG demonstrating the usage of the BranchPythonOperator."""
+"""Example DAG demonstrating the usage of the ``@task.branch`` TaskFlow API decorator."""
 
 import random
-from datetime import datetime
+from typing import List
```

Comment:
I wonder if this is a good chance to also switch these DAGs to `from __future__ import annotation`. It would allow us to write e.g. `list[str]` instead of importing from typing.

Reply:
Interesting idea. Since there is a _lot_ of blind copy/paste from examples, do we want `from __future__ import annotation` to be a pattern users should follow? This would be my only concern really.

Reply:
I think we are very close to releasing 2.4 and I think just upgrading everything to __future__ was a good idea - to make sure we have a nice cut-off point and to avoid cherry-picking problems to 2.3.2. **maybe** we are close to the cut-off point already?

Reply:
The `list[str]` syntax works out of the box on 3.9+ so the `from __future__ import annotation` line is not needed on those platforms. For 3.8 users, if they do blind copy-and-paste on partial code they get NameError due to `List` not being defined anyway, and if they know enough to copy the typing import, they probably can also notice the lack of `__future__` import.

Reply:
I am indifferent with that one. We already have a few cases where __future__ annotation is used (in `breeze` for example) so It's already spilled here and there in our codebase. It's harmless for cherry-picking in examples, so this is not a worry for me. I guess we can slowly start with adding __future_ annotations in places where we have low probability of conflicts with cherry-picking.

Reply:
OK, I have no problem adding it to these examples. Happened to find this [flake8 plugin](https://pypi.org/project/flake8-future-annotations/) for future annotations if we want to start enforcing it in CI too in, dare I say, the future.

Reply:
I don’t think we want to _enforce_ it for now. We can change each file gradually when we hit them instead, and pyupgrade performs per-file checks quite well already (it enforces modern syntax if you add the `__future__` import in a file).